lib/engine.py

```diff
@@ -288,8 +284,10 @@ def convert_data_type(self, datatype):
             thistype = ""
 
         if thispk:
-            thistype = self.pkformat % thistype
-
+            if isinstance(thistype, tuple):
+                thistype = self.pkformat % thistype
+            else:
+                thistype = self.pkformat % (thistype, "")
```

Comment:
This if/else is basically a check on the length of `thistype` right? The first line works if it's 2 items, the second if it's only 1. We have a similar check above.

I guess in general I'm wondering if we really need three separate sets of if/else clauses that all set `thistype`. It seems like it would be clearer to have one big if/else clause, potentially with some compound conditions (i.e., conditionals that use `and` or `or`). That way it's clear that that chunk of code is being used to set `thistype` based on the conditions.